addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/CacheableScannerUnitTest.java

```diff
@@ -79,4 +116,205 @@ public void shouldNotCauseExceptionWhenExpiresHeaderHasZeroValue()
         // Then
         assertThat(alertsRaised.size(), equalTo(1));
     }
+
+    @Test
+    public void shouldRaiseAlertStoreAndCacheableWhenStatusNonCacheableByDefaultAndExpiryDateGiven()
+            throws URIException, HttpMalformedHeaderException {
```

Comment:
`URIException` can be thrown in this class by `requestHeader.setURI(new URI("https://example.com/fred/", false));` in line 39.
I added the URI to the request header, to prevent `NullPointerException` when setting the alert details in method `CacheableScanner.alertStoreableCacheable` in line 860:
`msg.getRequestHeader().getURI().toString()`.

Reply:
As with the other PR not a major issue, I just couldn’t recall how other unit tests were handling it and it seemed out of place.